GlobalstatsIO.cs

```diff
@@ -5,318 +5,315 @@
-
-		WWW www = new WWW (url, pData, headers);
-
-		while (!www.isDone) {
-			System.Threading.Thread.Sleep (100);
-		}
-
-		// check for errors
-		if (www.error == null) {
-			//UnityEngine.Debug.Log ("WWW POST Ok!");
-		} else {
-			UnityEngine.Debug.Log ("WWW POST Error: " + www.error);
-			UnityEngine.Debug.Log ("WWW POST Content: " + www.text);
-			return false;
-		}
-
-		link_data = JsonUtility.FromJson<GlobalstatsIO_LinkData> (www.text);
-		return true;
-	}
-	
-	// numberOfPlayer can be 100 at max.
-	public GlobalstatsIO_Leaderboard getLeaderboard(string gtd, int numberOfPlayers) {
-
-
-		if (numberOfPlayers < 0) {
-			return new GlobalstatsIO_Leaderboard();
-		} else if(numberOfPlayers > 100) { // Number has to be between 0 and 100
-			numberOfPlayers = 100;
-		}
-
-		if (api_access_token == null || !api_access_token.isValid())
-		{
-			if (!getAccessToken())
-			{
-				return new GlobalstatsIO_Leaderboard();
-			}
-		}
-
-
-		string url = "https://api.globalstats.io/v1/gtdleaderboard/" + gtd;
-
-		string json_payload = "{\"limit\":" + numberOfPlayers + "\n}";
-
-		Dictionary<string, string> headers = new Dictionary<string, string>();
-		headers.Add("Authorization", "Bearer " + api_access_token.access_token);
-		headers.Add("Content-Type", "application/json");
-		headers.Add("Cache-Control", "no-cache");
-		headers.Add("Content-Length", json_payload.Length.ToString());
-
-		byte[] pData = Encoding.ASCII.GetBytes(json_payload.ToCharArray());
-
-
-		WWW www = new WWW(url, pData, headers);
-
-
-		while (!www.isDone)
-		{
-			System.Threading.Thread.Sleep(100);
-		}
-
-		// check for errors
-		if (www.error == null)
-		{
-			//UnityEngine.Debug.Log ("WWW POST Ok!");
-		}
-		else
-		{
-			UnityEngine.Debug.Log("WWW POST Error: " + www.error);
-			UnityEngine.Debug.Log("WWW POST Content: " + www.text);
-			return new GlobalstatsIO_Leaderboard();
-		}
-
-		GlobalstatsIO_Leaderboard leaderboard = JsonUtility.FromJson<GlobalstatsIO_Leaderboard>(www.text);
-
-		return leaderboard;
-	}
-	
-}
+    [Serializable]
+    public class StatisticValues
+    {
+        public string key = null;
+        public string value = "0";
+        public string sorting = null;
+        public string rank = "0";
+        public string value_change = "0";
+        public string rank_change = "0";
+    }
+
+    [Serializable]
+    public class LinkData
+    {
+        public string url = null;
+        public string pin = null;
+    }
+
+    [Serializable]
+    public class LeaderboardValue
+    {
+        public string name = null;
+        public string user_profile = null;
+        public string user_icon = null;
+        public string rank = "0";
+        public string value = "0";
+    }
+
+    [Serializable]
+    public class Leaderboard
+    {
+        public LeaderboardValue[] data;
+    }
+
+    public class GlobalstatsIOClient
+    {
+        [HideInInspector]
+        public static string api_id = "";
```

Comment:
Any reason this need to be public?

Reply:
They way the previous implementation was written, this was public and set outside of the class using the static string. I'll make it private and settable only through the constructor. That makes more sense than having just these as static.